pkg/cluster/registerWithHive.go

```diff
@@ -0,0 +1,127 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"os"
+	"time"
+
+	hive "github.com/Azure/ARO-RP/pkg/hiveregistrator"
+	"github.com/openshift/installer/pkg/asset/installconfig/azure"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (m *manager) collectDataForHive(ctx context.Context) error {
+
+	m.log.Print("collecting registration data from new cluster")
+
+	cv, err := m.configcli.ConfigV1().ClusterVersions().Get(ctx, "version", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	clusterid := cv.Spec.ClusterID
+
+	servicePrincipal := azure.Credentials{
+		TenantID:       m.env.TenantID(),
+		SubscriptionID: m.env.SubscriptionID(),
+		ClientID:       m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.ClientID,
+		ClientSecret:   string(m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.ClientSecret),
+	}
+
+	spBytes, err := json.Marshal(servicePrincipal)
+	if err != nil {
+		return err
+	}
+
+	// TODO: What if a customer adds/modifies a pullsecret after the install ?
+	// TODO: All parameter locations to be verified
+	m.hiveWorkloadCluster = &hive.WorkloadCluster{
+		Subscription:                m.subscriptionDoc.ID,
+		Name:                        m.doc.OpenShiftCluster.Name,
+		Location:                    m.doc.OpenShiftCluster.Location,
+		BaseDomain:                  m.env.Domain(),
+		BaseDomainResourceGroupName: m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID,
+		InfraID:                     m.doc.OpenShiftCluster.Properties.InfraID,
+		ClusterID:                   string(clusterid),
+		KubeConfig:                  string(m.doc.OpenShiftCluster.Properties.AdminKubeconfig),
+		PullSecret:                  string(m.doc.OpenShiftCluster.Properties.ClusterProfile.PullSecret),
+		ServicePrincipal:            string(spBytes),
+	}
+
+	return nil
+
+}
+
+func (m *manager) initializeHiveClusterClients(ctx context.Context) error {
+	// don't fail the installation on hive related errors
+	kubeConfigPath := os.Getenv("HIVEKUBECONFIGPATH")
```

Comment:
We need to get rid of `os.Getenv` call here. Please try to move it as close as possible to the command entrypoint.

Reply:
I'm aware of that, will fix later.. For now it's just more convenient to be able to change the env variable without having to restart the RP